homeassistant/components/renault/binary_sensor.py

```diff
@@ -59,20 +71,40 @@ class RenaultBinarySensor(
     @property
     def is_on(self) -> bool | None:
         """Return true if the binary sensor is on."""
+
+        if self.entity_description.value_lambda is not None:
+            return self.entity_description.value_lambda(self)
+        if self.entity_description.on_key is None:
+            raise NotImplementedError("Either value_lambda or on_key must be set")
         if (data := self._get_data_attr(self.entity_description.on_key)) is None:
             return None
 
         return data == self.entity_description.on_value
 
 
+def _plugged_in_value_lambda(self: RenaultBinarySensor) -> bool | None:
+    """Return true if the vehicle is plugged in."""
+
+    data = self.coordinator.data
+    plug_status = data.get_plug_status() if data else None
+
+    if plug_status is not None:
+        return plug_status == PlugState.PLUGGED
+
+    charging_status = data.get_charging_status() if data else None
+    if charging_status is not None and charging_status in _PLUG_FROM_CHARGE_STATUS:
+        return True
+
+    return None
```

Comment:
Another solution might be to make _PLUG_FROM_CHARGE_STATUS a dict[ChargeState | None, bool]:
```suggestion
    return _PLUG_FROM_CHARGE_STATUS.get(charging_status)
```